bin/full_conversion.py

```diff
@@ -0,0 +1,49 @@
+
+import argparse
+import ROOT
+
+def main():
+    parser = argparse.ArgumentParser(description='Convert detector')
+    parser.add_argument('-c', '--compact', help='Compact file location(s)',
+                        required=True, type=str, nargs='+')
+    parser.add_argument('-o', '--out', help='Converted file path',
+                        default='nothing', type=str)
+    parser.add_argument('-v', '--visibility', help='Level of layers in detector',
+                        default=9, type=int)
+
+    args = parser.parse_args()
+
+    convert(args.compact, args.out, args.visibility)
+
+
+def convert(compact_files, out_path, visibility):
+    print('INFO: Converting following compact file(s):')
+    for cfile in compact_files:
+        print('      ' + cfile)
+
+    ROOT.gSystem.Load('libDDCore')
+    description = ROOT.dd4hep.Detector.getInstance()
+    for cfile in compact_files:
+        description.fromXML(cfile)
+
+    ROOT.gGeoManager.SetVisLevel(visibility)
+    ROOT.gGeoManager.SetVisOption(0)
+
+    if out_path == 'nothing':
+
+        counter = 0
+        slash_list = []
+        for cfile in compact_files:
+            for i in cfile:
+                counter += 1
+                if i == '/':
+                    slash_list.append(counter)
+
+            last_slash = max(slash_list)
```

Comment:
I would move this into a function to declutter the conversion function. I am also not entirely sure I understand what the output will be yet if there are multiple compact files.

The end result would look something like
```python
    # ... all the rest from above
    ROOT.gGeoManager.SetVisOption(0)

    out_path = determine_outpath(out_path, compact_files)  # to be implemented
    ROOT.gGeoManager.Export(out_path)
```